### PR TITLE
BUG: convert_dtypes ingoring convert keywords for pyarrow backend

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

cc @mroeschke 
This seems buggy, we should still respect the keywords

### CODE REVIEW

File: pandas/core/dtypes/cast.py [hunk 1]
```diff
@@ -1106,20 +1106,30 @@ def convert_dtypes(
...
-            # variable has type "Union[dtype[Any], ExtensionDtype, None]")
-            base_dtype = inferred_dtype  # type: ignore[assignment]
-        pa_type = to_pyarrow_type(base_dtype)
-        if pa_type is not None:
-            inferred_dtype = ArrowDtype(pa_type)
+        if (
+            (convert_integer and inferred_dtype.kind in "iu")
+            or (convert_floating and inferred_dtype.kind in "fc")
+            or (convert_boolean and inferred_dtype.kind == "b")
+            or (convert_string and is_string_dtype(inferred_dtype))
```
Comment (mroeschke) [inline pandas/core/dtypes/cast.py:1113]:
```suggestion
            or (convert_string and isinstance(inferred_dtype, StringDtype)
```
?

File: pandas/core/dtypes/cast.py [hunk 2]
```diff
@@ -1106,20 +1106,30 @@ def convert_dtypes(
...
+            or (
+                inferred_dtype.kind not in "iufcb"
+                and not is_string_dtype(inferred_dtype)
```
Comment (mroeschke) [inline pandas/core/dtypes/cast.py:1116]: Same here potentially

**Comment by mroeschke** [comment]
```
mypy.....................................................................Failed
- hook id: mypy
- duration: 163.23s
- exit code: 1

pandas/core/dtypes/cast.py:1110: error: Item "str" of "Union[str, Any, dtype[Any], ExtensionDtype]" has no attribute "kind"  [union-attr]
pandas/core/dtypes/cast.py:1111: error: Item "str" of "Union[str, Any, dtype[Any], ExtensionDtype]" has no attribute "kind"  [union-attr]
pandas/core/dtypes/cast.py:1112: error: Item "str" of "Union[str, Any, dtype[Any], ExtensionDtype]" has no attribute "kind"  [union-attr]
pandas/core/dtypes/cast.py:1115: error: Item "str" of "Union[str, Any, dtype[Any], ExtensionDtype]" has no attribute "kind"  [union-attr]
Found 4 errors in 1 file (checked 1417 source files)
```

**Comment by phofl** [comment]
thx fixed

**Comment by mroeschke** [comment]
Merge conflict otherwise looks good

**Pull request merged**